TestHelpers.Tests/MockFileCopyTests.cs

```diff
@@ -191,6 +213,50 @@ public void MockFile_Copy_ShouldThrowNotSupportedExceptionWhenTargetFileNameCont
             }
         }
 
+        [Test]
+        [WindowsOnly(WindowsSpecifics.Drives)]
+        public void MockFile_Copy_ShouldThrowNotSupportedExceptionWhenSourcePathContainsInvalidUseOfDriveSeparator()
+        {
+            var badSourcePath = XFS.Path(@"C::\something\demo.txt");
+            var destinationPath = XFS.Path(@"C:\elsewhere\demo.txt");
+            var fileSystem = new MockFileSystem();
+
+            Assert.Throws<NotSupportedException>(() => fileSystem.File.Copy(badSourcePath, destinationPath));
+        }
+
+        [Test]
+        [WindowsOnly(WindowsSpecifics.Drives)]
+        public void MockFile_Copy_ShouldThrowNotSupportedExceptionWhenSourcePathContainsInvalidDriveLetter()
+        {
+            var badSourcePath = XFS.Path(@"0:\something\demo.txt");
+            var destinationPath = XFS.Path(@"C:\elsewhere\demo.txt");
+            var fileSystem = new MockFileSystem();
+
+            Assert.Throws<NotSupportedException>(() => fileSystem.File.Copy(badSourcePath, destinationPath));
+        }
+
+        [Test]
+        [WindowsOnly(WindowsSpecifics.Drives)]
```

Comment:
Would this necessarily be a Windows only feature? I may need to look into it more, but the C:\ path will just be converted to a unix path, along the lines of /tmp/some/dir, and I would think /elsewhere:/demo.txt would be a valid case and something to test against a Unix environment.

Tests verify behavior, skipping them on an environment doesn't prevent the behavioral changes from going through, they just make us blind to the side effects.

Reply:
According to the docs for [Path.VolumeSeparatorChar](https://msdn.microsoft.com/en-us/library/system.io.path.volumeseparatorchar(v=vs.110).aspx):

> The value of this field is a colon (:) on Windows and Macintosh, and a slash (/) on UNIX operating systems. This is most useful for parsing paths such as "c:\windows" or "MacVolume:System Folder".

So on Unix, the test would just be inserting a `/`, which wouldn't cause any exception to be raised.

`Drives` isn't specifically the reason why the test is Windows-only, but Unix doesn't have an analog to the `:` that Windows uses to separate drive letters from the rest of the path. Could add `WindowsSpecifics.VolumeSeparator`?

Reply:
I think this approach is fine. @jpreese Any objections left?